### PR TITLE
Add custom API key settings section

### PR DESCRIPTION
**Pull request by prestosole**

### CODE REVIEW

File: packages/clima_ui/lib/screens/loading_screen.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
```
Comment (prestosole) [inline packages/clima_ui/lib/screens/loading_screen.dart]: I guess this file sneaked in when we tried to rebase the PR

File: packages/clima_ui/lib/screens/settings_screen.dart [hunk 2]
```diff
@@ -111,6 +118,56 @@ class SettingScreen extends ConsumerWidget {
               ),
             ),
             const SettingsDivider(),
+            const SettingsHeader(
+              title: 'API key',
+            ),
```
Comment (triallax) [inline packages/clima_ui/lib/screens/settings_screen.dart]:
```suggestion
            const SettingsHeader(title: 'API key'),
```

File: packages/clima_ui/lib/screens/settings_screen.dart [hunk 3]
```diff
@@ -111,6 +118,56 @@ class SettingScreen extends ConsumerWidget {
...
+              leading: Icon(
+                Icons.restore_outlined,
+                color: Theme.of(context).iconTheme.color,
+              ),
+              onTap: () {
+                showDialog(
+                  context: context,
+                  builder: (context) {
+                    return const ApiKeyResetDialog();
+                  },
```
Comment (triallax) [inline packages/clima_ui/lib/screens/settings_screen.dart]: Use fat arrow lambda here.

File: packages/clima_ui/lib/screens/settings_screen.dart [hunk 4]
```diff
@@ -111,6 +118,56 @@ class SettingScreen extends ConsumerWidget {
...
+              leading: Icon(
+                Icons.launch_outlined,
+                color: Theme.of(context).iconTheme.color,
+              ),
+              onTap: () {
+                showDialog(
+                  context: context,
+                  builder: (context) {
+                    return const ApiKeyInfoDialog();
+                  },
```
Comment (triallax) [inline packages/clima_ui/lib/screens/settings_screen.dart]: Same as above.

File: packages/clima_ui/lib/screens/weather_screen.dart [hunk 5]
```diff
@@ -43,7 +47,12 @@ class WeatherScreen extends HookConsumerWidget {
 
     useEffect(
       () {
-        Future.microtask(() => fullWeatherStateNotifier.loadFullWeather());
+        Future.microtask(
+          () => Future.wait([
+            apiKeyStateNotifier.loadApiKey(),
+            fullWeatherStateNotifier.loadFullWeather()
```
Comment (triallax) [inline packages/clima_ui/lib/screens/weather_screen.dart]:
```suggestion
            fullWeatherStateNotifier.loadFullWeather(),
```

File: packages/clima_ui/lib/utilities/snack_bars.dart [hunk 6]
```diff
@@ -44,3 +44,29 @@ void showFailureSnackBar(
     ),
   );
 }
+
+void showSnackBar(
```
Comment (triallax) [inline packages/clima_ui/lib/utilities/snack_bars.dart:40]: `showFailureSnackBar` should be refactored to use this.

File: packages/clima_ui/lib/widgets/dialogs/api_key/api_key_reset_dialog.dart [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+
+class ApiKeyResetDialog extends StatelessWidget {
+  const ApiKeyResetDialog({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
```
Comment (triallax) [inline packages/clima_ui/lib/widgets/dialogs/api_key/api_key_reset_dialog.dart]: Use fat arrow syntax here.

**Review by triallax — CHANGES_REQUESTED** [review]

**Review by triallax — APPROVED** [review]
LGTM, you can merge when the CI goes green.

**Pull request merged**